File: Sort/Longest Word in Dictionary through Deleting/LongestWordinDictionarythroughDeleting.js
/**
 * @param {string} s
 * @param {string[]} d
 * @return {string}
 */
var findLongestWord = function(s, d) {
  let ans = ''
  for (let j = 0; j < d.length; j++) {
      let idx = 0
      for (let i = 0; i < s.length; i++) {
          if (idx == d[j].length)
              break
          if (s[i] == d[j][idx])
              idx += 1
      }
      if (idx != d[j].length)
          continue
      if (ans.length < d[j].length || (ans.length == d[j].length && d[j] < ans))
          ans = d[j]
  }
  return ans
};

File: Sort/Longest Word in Dictionary through Deleting/LongestWordinDictionarythroughDeleting.cpp
class Solution
{
public:
  string findLongestWord(string s, vector<string> &d)
  {
    string ans;
    for (int i = 0; i < d.size(); i++)
    {
      int idx = 0;
      for (int j = 0; j < s.size(); j++)
      {
        if (idx == d[i].size())
          break;
        if (s[j] == d[i][idx])
          idx += 1;
      }
      if (idx != d[i].size())
        continue;
      if (ans.size() < d[i].size() || (ans.size() == d[i].size() && d[i] < ans))
        ans = d[i];
    }
    return ans;
  }
};